tests/components/reolink/test_init.py

```diff
@@ -70,11 +64,33 @@ async def test_failures_parametrized(
     assert config_entry.state == expected
 
 
+async def test_firmware_error_twice(
+    hass: HomeAssistant,
+    reolink_connect: MagicMock,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test when the firmware update fails 2 times."""
+    reolink_connect.check_new_firmware = AsyncMock(
+        side_effect=ReolinkError("Test error")
+    )
+    assert await hass.config_entries.async_setup(config_entry.entry_id) is True
+    await hass.async_block_till_done()
+
+    assert config_entry.state == ConfigEntryState.LOADED
+
+    firmware_coordinator = hass.data[const.DOMAIN][
+        config_entry.entry_id
+    ].firmware_coordinator
```

Comment:
We generally aren't supposed to access `hass.data` in test as its considered internal integration details

https://developers.home-assistant.io/docs/development_testing?_highlight=writing#writing-tests-for-integrations

Reply:
@bdraco alright I see, what I am trying to do is issue a second update of the firmware_coordinator (DataUpdateCoordinator), otherwise not all code is checked from the __init__.py file.

How can I request a second update from a DataUpdateCoordinator?

Reply:
Move time forward

Reply:
`async_fire_time_changed`

Reply:
But then I can't assume `last_update_success` of the DataUpdateCoordinator is first True and then False.
What schould I then assume as test?

Reply:
You can use caplog to check the logs

Reply:
Thank you for the help, I manged to make it work using `async_fire_time_change` and `hass.states.is_state`.